codegenerator/cli/templates/static/codegen/src/globalState/GlobalState.res

```diff
@@ -949,16 +952,25 @@ let injectedTaskReducer = (
             )
           dispatchAction(ErrorExit(errHandler))
         }
-      | {isInReorgThreshold: true}
+      | {isInReorgThreshold: true, val: None} =>
+        //pre registration is done, we've hit the multichain reorg threshold
+        //on the last batch and there are no items on the queue
+        dispatchAction(StartIndexingAfterPreRegister)
       | {val: None} if state.chainManager->ChainManager.isFetchingAtHead =>
-        //pre registration is done, start indexing
```

Comment:
This was a bug, I thought the second match arm of these 2 would have the "if" check just for this arm but it includes both when they both share the same return.